core/renderers/common/drawer.ts

```diff
@@ -440,4 +442,73 @@ export class Drawer {
       );
     }
   }
+
+  /** Returns a path to highlight the given connection. */
+  drawConnectionHighlightPath(conn: RenderedConnection) {
+    const measurable = this.info_.getMeasureableForConnection(conn);
+    if (!measurable) {
+      throw new Error('Could not find measurable for connection');
+    }
+
+    let path = '';
+    if (
+      conn.type === ConnectionType.INPUT_VALUE ||
+      conn.type === ConnectionType.OUTPUT_VALUE
+    ) {
+      path = this.getExpressionConnectionHighlightPath(measurable);
+    } else {
+      path = this.getStatementConnectionHighlightPath(measurable);
+    }
+    const block = conn.getSourceBlock();
+    if (block.pathObject.addConnectionHighlight) {
+      block.pathObject.addConnectionHighlight(
+        conn,
+        path,
+        conn.getOffsetInBlock(),
+        block.RTL,
+      );
+    }
```

Comment:
Can be simplified by use of optional chaining.  Will leave to your judgement as I'm not sure if it's more readable (though I suspect that with familiarity it will become so):
```suggestion
    block.pathObject.addConnectionHighlight?.(
      conn,
      path,
      conn.getOffsetInBlock(),
      block.RTL,
    );
```

Reply:
Huh I didn't know you could do optional chaining in that location. I thought it was only for dot referencing, not method calls. Sweet!